database/migrations/2022_11_06_201538_create_training_waiting_list_account_pending_removal_table.php

```diff
@@ -0,0 +1,37 @@
+<?php
+
+use Illuminate\Database\Migrations\Migration;
+use Illuminate\Database\Schema\Blueprint;
+use Illuminate\Support\Facades\Schema;
+
+return new class extends Migration
+{
+    /**
+     * Run the migrations.
+     *
+     * @return void
+     */
+    public function up()
+    {
+        Schema::create('training_waiting_list_account_pending_removal', function (Blueprint $table) {
+            $table->id();
+            $table->unsignedInteger('waiting_list_account_id');
+            $table->enum('status', ['Pending', 'Completed', 'Cancelled'])->default('Pending');
```

Comment:
The enum, for me, isn't serving any purpose since you're not tracking the changes between the statuses.

I think you'd be better off having columns like:

`pending_at`
`completed_at`
`cancelled_at` 

So you at least then have an audit trail. For querying those which are pending but not yet completed, you would get those where pending_at is not null, but completed at was (and further cancelled_at is null too).

Reply:
You could then convert these into methods on the model to save duplication and avoid having to reference the raw enum value each time. You then abstract away the underlying implementation leaving it more flexible in the future should you, say, want to change what constitutes pending